### ISSUE TITLE
responsivenessMetric requires Animations to have frameEvents, but some don't

### ISSUE THREAD

**Issue by surma** [catapult-project/catapult#2428]
When tracing a scroll and wanting to look at the responsiveness metric, I get this:
<img width="266" alt="screenshot 2016-06-24 15 35 08" src="https://cloud.githubusercontent.com/assets/234957/16340722/98a80c38-3a21-11e6-9972-907a748356ab.png">

cc @paulirish @paullewis

**Comment by eakuefner** [comment]
@benshayden

**Comment by benshayden** [comment]
Yeah, sorry. :-/
The right way to fix this would be to use either frame blame contexts or flow events or something to definitively match up the frame events to the AnimationExpectations. Scroll animations already support using flow events, but only if flow events are in the trace, and they're disabled by default because they blow up the trace file size. Frame blame contexts probably would not blow up the file size, but they might still be a bit buggy.

I had been hoping to come back to this after results2.html MVP and fix it the right way, but the quick-dirty solution would be to add a post-processing step to findInputExpectations to find all frame events that overlap AnimationExpectations, and if there aren't any, then delete the AnimationExpectation.

Can you link me to a benchmark or trace or other test case to make sure that the quick-dirty way would unblock you for now?

I can do that today or tomorrow, then figure out the frame blamer way after I return from vacation mid-August. Would that work for you?

**Comment by paulirish** [comment]
> Can you link me to a benchmark or trace or other test case to make sure that the quick-dirty way would unblock you for now?

I was seeing this for all traces with a scroll in desktop ToT.

**Comment by surma** [comment]
If this helps: The audit in lighthouse currently has the following trace flags

```
 [
      '-*', // exclude default
      'toplevel',
      'blink.console',
      'blink.user_timing',
      'benchmark',
      'devtools.timeline',
      'disabled-by-default-blink.debug.layout',
      'disabled-by-default-devtools.timeline',
      'disabled-by-default-devtools.timeline.frame',
      'disabled-by-default-devtools.timeline.stack',
      'disabled-by-default-devtools.screenshot'
]
```

and uses `chrome.gpuBenchmarking.smoothScrollBy()` to scroll.

**Comment by benshayden** [comment]
I think [this change](https://codereview.chromium.org/2115283002) should fix it, but I'm having trouble reproducing. Every time I scroll in a desktop chrome, the LatencyInfo flow events are in the benchmark category, so the existing InputLatencyAsyncSlice.associatedEvents finds the frame events.
Can one of you please either test that CL or send me a trace somehow?

**Comment by paulirish** [comment]
Here's a trace from desktop that triggers the screenshot above:  https://dl.dropboxusercontent.com/u/39519/rm-someday/trace_scrolling-on-paulirish_dot_com.json.gz

**Comment by benshayden** [comment]
Thanks! Yes, my change does fix responsivenessMetric. I'll submit it as soon as I get an lgtm, and fix any issues later.

**Comment by surma** [comment]
awesomesauce! Thanks a lot, @benshayden 👍

**Comment by paulirish** [comment]
This is fixed by 	2102c4b


all set for you now, surma
